command/agent/auth/cert/cert_test.go

```diff
@@ -0,0 +1,133 @@
+package cert
+
+import (
+	"context"
+	"os"
+	"path"
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/command/agent/auth"
+)
+
+func TestCertAuthMethod_Authenticate(t *testing.T) {
+	config := &auth.AuthConfig{
+		Logger:    hclog.NewNullLogger(),
+		MountPath: "cert-test",
+		Config: map[string]interface{}{
+			"name": "foo",
+		},
+	}
+
+	method, err := NewCertAuthMethod(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	client, err := api.NewClient(api.DefaultConfig())
```

Comment:
`api.DefaultConfig()` has an odd interface because it may set an error internally but not return it as a return value. Although it's unlikely in a test environment, I think we could miss errors by using the return value without checking.

Reply:
Good point. `NewClient` actually uses `DefaultConfig` internally if I don't pass in any, and also performs the error checking for us. I'll update this.